framework/src/loops/ComputeUserObjectsThread.C

```diff
@@ -191,6 +198,51 @@ ComputeUserObjectsThread::onInternalSide(const Elem * elem, unsigned int side)
   }
 }
 
+void
+ComputeUserObjectsThread::onInterface(const Elem * elem, unsigned int side, BoundaryID bnd_id)
+{
+
+  // Pointer to the neighbor we are currently working on.
+  const Elem * neighbor = elem->neighbor_ptr(side);
+
+  // Get the global id of the element and the neighbor
+  const dof_id_type elem_id = elem->id(), neighbor_id = neighbor->id();
+
+  if (!_interface_user_objects.hasActiveBoundaryObjects(bnd_id, _tid))
+  {
+    return;
+  }
+
+  if (!((neighbor->active() && (neighbor->level() == elem->level()) && (elem_id < neighbor_id)) ||
+        (neighbor->level() < elem->level())))
+  {
+    return;
+  }
+
+  _fe_problem.prepareFace(elem, _tid);
+  // TODO next line does not produce nay effect
+  // _fe_problem.reinitElemFace(elem, side, bnd_id, _tid);
+  _fe_problem.reinitNeighbor(elem, side, _tid);
+
+  // Set up Sentinels so that, even if one of the reinitMaterialsXXX() calls throws, we
+  // still remember to swap back during stack unwinding.
+  SwapBackSentinel face_sentinel(_fe_problem, &FEProblem::swapBackMaterialsFace, _tid);
+  _fe_problem.reinitMaterialsFace(elem->subdomain_id(), _tid);
+
+  // TODO activating the next two lines generate a segmentation fault, need to understand why
```

Comment:
In `onBoundary` there is only one sentinel: the lines of code are:
```
  SwapBackSentinel sentinel(_fe_problem, &FEProblem::swapBackMaterialsFace, _tid);
  _fe_problem.reinitMaterialsFace(_subdomain, _tid);
  _fe_problem.reinitMaterialsBoundary(bnd_id, _tid);
```

Reply:
Perhaps that would resolve the segmentation fault?

Reply:
I tried to use was you suggested but it still result in a segfault. 
`_fe_problem.reinitMaterialsBoundary(bnd_id, _tid);` always cause a segfault. I wondering if we really need it, as the interface UO should not update the value of the boundary material property but only hold the quantities that will be used in computeQP

Reply:
@lindsayad  I went to the bottom of this and found what's happening:
The `reinitMaterialsBoundary` call for update stateful material boundary property, however the UO itself must be executed to update the stateful interface MP... therefore there is no value that can be pulled from the UO. The idea here is that stateful Interface MP are function of stateful and non stateful boundary and bulk properties, and should always be updated at the end. I think we might need to prevent the update of stateful interface MP.